### DISCUSSION TITLE
Slides to "sell" VTL to business users

### DISCUSSION THREAD

**Discussion by brainwasher** [VTL-Community/VTL-Community#37] [Q&A]
Dears,

we are currently looking for some inspiration for slides that we can use to sell VTL to business users (statisticians, middle managers from business units). There are still some concerns about using VTL as compared to SQL, Python... and some lack of understanding on main benefits for business to have a standardised non-IT syntax and also the functional approach of VTL.

Do you have something (or have seen something) that we can use as starting point for our internal stakeholder management? It would be great! We'd also be happy to share the result of whatever we make of it :) 

Thanks,
Daniel.

**Comment by antonio-olleros** [comment]
Hi all,

I wrote some notes some time ago, here they are:

VTL has been designed with a very specific use case in mind: Regulatory reporting/statistics. The use case implies two main requirement: User-friendliness and executability in any platform. 
Regarding user friendliness, the main requirements are that business users should be able to write and to understand VTL statements.
Regarding executability, the main requirements is that the language is formal, so that it is executable, and that it is technology agnostic, so that it can be  executed by any institution regardless of issues like operating systems or DB  technologies.
 
So the advantages of VTL as compared to other languages shall be evaluated on those terms. 
Comparing to other languages, we believe that the main alternative candidate would be SQL. Python or R would be other candidates, but, on top of most of the disadvantages SQL have, they are much more generalist languages, scripting languages, which would make them very difficult to standardise and make understandable.
 
Technology
•	VTL is technology agnostic while SQL is not. SQL is standard to a certain point. Many relevant operators are specific to the RDBMS (Oracle, SQL Server…). Besides standard SQL is not procedural. For instance, PL/SQL is the Procedural SQL specific for Oracle. So any procedural SQL is specific to a technology and cannot be run in other databases.
•	SQL depends on an underlying data model. Changes in the way information is stored implies changes in the SQL required to achieve the same result.
•	VTL is functional, which enables the visualisation of data lineage and the generation of exports that facilitate the understanding by business users.
•	VTL provides specific artifacts for data validation (check operator, datapoint rulesets and hierarchical rulesets) which greatly facilitates writing validations and standardising the way validations are written and presented.
 
User friendliness
•	SQL becomes extremely complex to write and understand for very common use cases in statistics. Just think of how you would write a simple validation like total for the EU has to be equal to the sum of the EU countries.
•	VTL enables the visualisation of transformation graphs, which facilitates understanding by business users.
•	VTL provides a standard way to do things, e.g., validations in VTL are always written the same way. In SQL (even more in Python or R) there are many ways to do the same thing. This complicates writing and, specially, understanding the transformations, since different people are likely to write very different things

**Reply by brainwasher** [reply]
Thanks Antonio!

**Comment by amattioc** [comment]
I would also add two more things:

- when we talk about R or Python or SQL we always need to specify flavour, version, add-ons (and add-ons versions) and this greatly reduces the possibilities of standardization
- VTL provides a clear, metadata driven, behaviour for basic but key operators. The generic languages do not. Just think about a simple sum of two datasets. In VTL you know exactly what will happen. In R or python it will trigger an error unless you overload it in a way that, using attributes, will correctly handle identifiers and measures. But this basically means to reimplement parts of VTL in R or Python....

**Comment by antonio-olleros** [comment]
I agree with Attilio's addition!

BTW, I think it would be very useful to show this with real examples. We could prepare a real and normal case in VTL, and see how it would look like in SQL, Python (Pandas, I guess) and R. I would be happy to contribute, if you think it is useful
